connector/microsoft/microsoft.go

```diff
@@ -299,37 +320,40 @@ type group struct {
 	Name string `json:"displayName"`
 }
 
-func (c *microsoftConnector) getGroups(ctx context.Context, client *http.Client, userID string) (groups []string, err error) {
-	ids, err := c.getGroupIDs(ctx, client)
+func (c *microsoftConnector) getGroups(ctx context.Context, client *http.Client, userID string) ([]string, error) {
+	userGroups, err := c.getGroupIDs(ctx, client)
 	if err != nil {
-		return groups, err
+		return nil, err
 	}
 
-	groups, err = c.getGroupNames(ctx, client, ids)
-	if err != nil {
-		return
+	if c.groupNameFormat == GroupName {
+		userGroups, err = c.getGroupNames(ctx, client, userGroups)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	// ensure that the user is in at least one required group
-	isInGroups := false
 	if len(c.groups) > 0 {
+		filteredGroups := make([]string, 0)
 		gs := make(map[string]struct{})
 		for _, g := range c.groups {
 			gs[g] = struct{}{}
 		}
 
-		for _, g := range groups {
+		for _, g := range userGroups {
 			if _, ok := gs[g]; ok {
-				isInGroups = true
-				break
+				filteredGroups = append(filteredGroups, g)
 			}
 		}
-	}
-	if len(c.groups) > 0 && !isInGroups {
-		return nil, fmt.Errorf("microsoft: user %v not in required groups", userID)
+
+		if len(filteredGroups) == 0 {
+			return nil, fmt.Errorf("microsoft: user %v not in required groups", userID)
+		}
+		return filteredGroups, nil
```

Comment:
Returning a subset of groups is a breaking change. Please revert this logic or use a flag.

Reply:
🤦‍♂ 
thanks, I added a flag `useGroupsAsWhitelist`